src/test/java/seedu/trackermon/model/show/NameContainsKeywordsPredicateTest.java

```diff
@@ -0,0 +1,75 @@
+package seedu.trackermon.model.show;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.trackermon.testutil.ShowBuilder;
+
+public class NameContainsKeywordsPredicateTest {
+
+    @Test
+    public void equals() {
+        List<String> firstPredicateKeywordList = Collections.singletonList("first");
+        List<String> secondPredicateKeywordList = Arrays.asList("first", "second");
+
+        NameContainsKeywordsPredicate firstPredicate = new NameContainsKeywordsPredicate(firstPredicateKeywordList);
+        NameContainsKeywordsPredicate secondPredicate = new NameContainsKeywordsPredicate(secondPredicateKeywordList);
+
+        // same object -> returns true
+        assertTrue(firstPredicate.equals(firstPredicate));
+
+        // same values -> returns true
+        NameContainsKeywordsPredicate firstPredicateCopy = new NameContainsKeywordsPredicate(firstPredicateKeywordList);
+        assertTrue(firstPredicate.equals(firstPredicateCopy));
+
+        // different types -> returns false
+        assertFalse(firstPredicate.equals(1));
+
+        // null -> returns false
+        assertFalse(firstPredicate.equals(null));
+
+        // different name -> returns false
```

Comment:
Hmmm, I was wondering if we should standard where we place the comments. AB3 places theirs at the back of the statement. I have yet to to it on mine as well. Maybe, we should raise a issue so that we can standardise this?

Reply:
I've noticed AB3 has put comments both before and at the back of the statement. The comment placed in the front is to tell the developer the purpose of what the statement is testing for. While the comment placed at the back of the statement is just a description of what the statement is. May I know what are your thoughts on this @AY2122S2-CS2103T-T09-3/developers

Reply:
Erm i don't understand and didn't see any comment at the back. Are there any examples?

Reply:
Anyway, the coding standard says comments should be indented relative to their position in the code.
![image](https://user-images.githubusercontent.com/77216013/162201282-368bebd2-58ab-4bce-9e6f-5d7f30053527.png)

So if you put a comment right behind the statement in the same line, it is technically violating the standard as it is not indented to the same position as the code.

Reply:
I think what weijun might is referring to is comments within the same line as the statement. For example,
```
// tests methods A, B, C
TestA();
TestB();
TestC(); // test c should fail
```
The comment `// tests methods A, B, C` is describing the section below that it is testing for the methods below whereas `// test c should fail` is describing the statement of what it does.

> I've noticed AB3 has put comments both before and at the back of the statement. The comment placed in the front is to tell the developer the purpose of what the statement is testing for. While the comment placed at the back of the statement is just a description of what the statement is. May I know what are your thoughts on this @AY2122S2-CS2103T-T09-3/developers

What @Ardentsoul mentioned regarding commenting on top and within the statement is correct. AB3 if I recalled uses both kind of comments. I think there is no need to standardize anything as this is something correct. 

What @lowkaiwei98 mentioned is that the indentation of the comment must be on the same line with the section it is referring to.